homeassistant/components/almond/__init__.py

```diff
@@ -0,0 +1,230 @@
+
+    host = conf[CONF_HOST]
+
+    if conf[CONF_TYPE] == TYPE_OAUTH2:
+        config_flow.AlmondFlowHandler.async_register_implementation(
+            hass,
+            config_entry_oauth2_flow.LocalOAuth2Implementation(
+                hass,
+                DOMAIN,
+                conf[CONF_CLIENT_ID],
+                conf[CONF_CLIENT_SECRET],
+                f"{host}/me/api/oauth2/authorize",
+                f"{host}/me/api/oauth2/token",
+            ),
+        )
+        return True
+
+    if not hass.config_entries.async_entries(DOMAIN):
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": config_entries.SOURCE_IMPORT},
+                data={"type": TYPE_LOCAL, "host": conf[CONF_HOST]},
+            )
+        )
+    return True
+
+
+async def async_setup_entry(hass, entry):
+    """Set up Almond config entry."""
+    websession = aiohttp_client.async_get_clientsession(hass)
+    if entry.data["type"] == TYPE_LOCAL:
+        auth = AlmondLocalAuth(entry.data["host"], websession)
+
+    else:
+        # OAuth2
+        implementation = await config_entry_oauth2_flow.async_get_config_entry_implementation(
+            hass, entry
+        )
+        oauth_session = config_entry_oauth2_flow.OAuth2Session(
+            hass, entry, implementation
+        )
+        auth = AlmondOAuth(entry.data["host"], websession, oauth_session)
+
+    api = WebAlmondAPI(auth)
+    agent = AlmondAgent(api)
+
+    # Hass.io does its own configuration of Almond.
+    if entry.data.get("is_hassio"):
+        conversation.async_set_agent(hass, agent)
+        return True
+
+    # Configure Almond to connect to Home Assistant
+    store = storage.Store(hass, STORAGE_VERSION, STORAGE_KEY)
+    data = await store.async_load()
+
+    if data is None:
+        data = {}
+
+    user = None
+    if "almond_user" in data:
+        user = await hass.auth.async_get_user(data["almond_user"])
+
+    if user is None:
+        user = await hass.auth.async_create_system_user("Almond", [GROUP_ID_ADMIN])
+        data["almond_user"] = user.id
+        await store.async_save(data)
+
+    refresh_token = await hass.auth.async_create_refresh_token(
+        user,
+        # Almond will be fine as long as we restart once every 5 years
+        access_token_expiration=timedelta(days=365 * 5),
+    )
+
+    # Create long lived access token
+    access_token = hass.auth.async_create_access_token(refresh_token)
+
+    # Store token in Almond
+    try:
+        with async_timeout.timeout(10):
+            await api.async_create_device(
+                {
+                    "kind": "io.home-assistant",
+                    "hassUrl": hass.config.api.base_url,
+                    "accessToken": access_token,
+                    "refreshToken": "",
+                    # 5 years from now in ms.
+                    "accessTokenExpires": time.time() + (60 * 60 * 24 * 365 * 5) * 1000,
```

Comment:
The math here is wrong: it should be `(time.time() + 60 * 60 * 24 * 365 * 5) * 1000`